primefaces/src/main/java/org/primefaces/component/feedreader/RSSUtils.java

```diff
@@ -0,0 +1,111 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.primefaces.component.feedreader;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.primefaces.model.feedreader.FeedItem;
+import org.primefaces.util.Constants;
+
+import com.apptasticsoftware.rssreader.AbstractRssReader;
+import com.apptasticsoftware.rssreader.Item;
+import com.apptasticsoftware.rssreader.RssReader;
+import com.apptasticsoftware.rssreader.module.itunes.ItunesItem;
+import com.apptasticsoftware.rssreader.module.itunes.ItunesRssReader;
+
+/**
+ * Static utility class to parse feed input into a list of FeedItem objects.
+ */
+public class RSSUtils {
+
+    private RSSUtils() {
+        // Private constructor to prevent instantiation
+    }
+
+    /**
+     * Parses the input into a list of FeedItem objects.
+     *
+     * @param input     The input source, either a String representing the URL of the feed or an InputStream.
+     * @param size      The maximum number of items to retrieve from the feed.
+     * @param isPodcast Specifies whether the feed is a podcast feed.
+     * @return A list of FeedItem objects parsed from the input.
+     * @throws IOException If an I/O error occurs during the parsing process.
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    public static List<FeedItem> parse(Object input, int size, boolean isPodcast) throws IOException {
+        AbstractRssReader reader;
+        if (isPodcast) {
+            reader = new ItunesRssReader();
+        }
+        else {
+            reader = new RssReader();
+        }
+        if (input instanceof String) {
+            return convert((List<Item>) reader.read((String) input).limit(size).collect(Collectors.toList()));
+        }
+        return convert((List<Item>) reader.read((InputStream) input).limit(size).collect(Collectors.toList()));
```

Comment:
you can pass size to convert method, and stop the loop once limit reached

Reply:
But doesn't the "stream.limit(size)" already limit the size?

Reply:
Yeah, but if you have a huge feed, and you only want the first 10... Making me think, feed reader should be lazy-able (let's wait for someone to ask for it)

Reply:
I will check with the author of the lib he advertises he library handles huge feeds with speed so is he returning a proper stream that only reads as necessary from the upstream feed?

Reply:
Ohhh I see! My bad I thought the whole list was passed and then chunked... Not easy to read all these parenthesis on github review